nav2_common/nav2_common/launch/rewritten_yaml.py

```diff
@@ -129,15 +129,15 @@ def getYamlLeafKeys(self, yamlData):
 
   def convert(self, text_value):
     if self.__convert_types:
-      # try converting to float
+      # try converting to int
```

Comment:
Why change the order? I'd think you'd want float first since a float `2.0` could resolve as an `int` but more properly a float

Reply:
The thing is that "1665" ( the port ) would get converted to a float = 1665.0 . And this fails hard when I try to get the parameter via the .as_int() function.

In the setup before it is not possible to get a int from a string like "1665" as it also qualifies as float. 
https://stackoverflow.com/a/20929983

But now comes the bad thing I was not aware of.. 2.0 also qualifies as a int... So, int(2.3) works fine.. You just get 2 but no Exception is thrown. Damn it..

This would be working... `float(x) if '.' in x else int(x)` .. But I'm not sure this is qualifies as good enough..

Reply:
> The thing is that "1665" ( the port ) would get converted to a float = 1665.0 . And this fails hard when I try to get the parameter via the .as_int() function.

But you now introduce the problem the other way around, that's not going to fly. 

the `if '.'` thing would work. Better than now and there does have to be a `.` for it to resolve as a float regardless

Reply:
done